api/datadoghq/v2alpha1/datadogpodautoscaler_types.go

```diff
@@ -0,0 +1,560 @@
+
+	// DatadogPodAutoscalerLocalOwner states that this `DatadogPodAutoscaler` object is created/managed in Datadog app.
+	DatadogPodAutoscalerRemoteOwner DatadogPodAutoscalerOwner = "Remote"
+)
+
+// DatadogPodAutoscalerSpec defines the desired state of DatadogPodAutoscaler
+type DatadogPodAutoscalerSpec struct {
+	// TargetRef is the reference to the resource to scale.
+	TargetRef autoscalingv2.CrossVersionObjectReference `json:"targetRef"`
+
+	// Owner defines the source of truth for this object (local or remote)
+	// Value needs to be set when a DatadogPodAutoscaler object is created.
+	Owner DatadogPodAutoscalerOwner `json:"owner"`
+
+	// RemoteVersion is the version of the .Spec currently store in this object.
+	// Only set if the owner is Remote.
+	RemoteVersion *uint64 `json:"remoteVersion,omitempty"`
+
+	// Actuation defines how recommendations should be applied.
+	// +optional
+	// +kubebuilder:default={}
+	Actuation *DatadogPodAutoscalerActuation `json:"actuation,omitempty"`
+
+	// Objectives are objectives to reach and maintain for the target resource.
+	// Default to a single target to maintain 80% POD CPU utilization.
+	// +listType=atomic
+	// +optional
+	Objectives []DatadogPodAutoscalerObjective `json:"objectives,omitempty"`
+
+	// Constraints defines constraints that should always be respected.
+	Constraints *DatadogPodAutoscalerConstraints `json:"constraints,omitempty"`
+}
+
+// DatadogPodAutoscalerOwner defines the source of truth for this object (local or remote)
+// +kubebuilder:validation:Enum:=All;Manual;None
+type DatadogPodAutoscalerActuationMode string
+
+const (
+	// DatadogPodAutoscalerApplyApplyMode allows the controller to apply all recommendations (regular and manual)
+	DatadogPodAutoscalerApplyApplyMode DatadogPodAutoscalerActuationMode = "Apply"
+
+	// DatadogPodAutoscalerPreviewApplyMode doesn't allow the controller to apply any recommendations
+	DatadogPodAutoscalerPreviewApplyMode DatadogPodAutoscalerActuationMode = "Preview"
+)
+
+// DatadogPodAutoscalerActuation defines how recommendations should be applied.
+type DatadogPodAutoscalerActuation struct {
+	// Mode determines recommendations that should be applied by the controller:
+	// - All: Apply all recommendations (regular and manual).
+	// - Manual: Apply only manual recommendations (recommendations manually validated by user in the Datadog app).
+	// - None: Prevent the controller to apply any recommendations.
+	// It's also possible to selectively deactivate upscale, downscale or update actions thanks to the `Upscale`, `Downscale` and `Update` fields.
+	// +optional
+	// +kubebuilder:default=All
+	Mode DatadogPodAutoscalerActuationMode `json:"applyMode"`
+
+	// Update defines the policy to update target resource.
+	Update *DatadogPodAutoscalerUpdatePolicy `json:"update,omitempty"`
+
+	// Upscale defines the policy to scale up the target resource.
+	Upscale *DatadogPodAutoscalerScalingPolicy `json:"upscale,omitempty"`
```

Comment:
Wasn't there a proposal to change to scaleUp and scaleDown?

Reply:
Done